ARCs/arc-0025.md

```diff
@@ -0,0 +1,230 @@
+
+> For example, here is a standard WalletConnect URI:
+> ```
+> wc:4015f93f-b88d-48fc-8bfe-8b063cc325b6@1?bridge=https%3A%2F%2F9.bridge.walletconnect.org&key=b0576e0880e17f8400bfff92d4caaf2158cccc0f493dcf455ba76d448c9b5655
+> ```
+> And here is that same URI with the Algorand-specific query parameter:
+> ```
+> wc:4015f93f-b88d-48fc-8bfe-8b063cc325b6@1?bridge=https%3A%2F%2F9.bridge.walletconnect.org&key=b0576e0880e17f8400bfff92d4caaf2158cccc0f493dcf455ba76d448c9b5655&algorand=true
+> ```
+
+It is **RECOMMENDED** that dApps include this query parameter, but it is not **REQUIRED**. Wallets **MAY** reject sessions if the session URI does not contain this query parameter.
+
+#### Chain IDs
+
+WalletConnect v1 sessions are associated with a numeric chain ID. Since Algorand chains do not have numeric identifiers (instead, the genesis hash or ID is used for this purpose), this document defines the following chain IDs for the Algorand ecosystem:
+
+* MainNet (genesis hash `wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=`): 416001
+* TestNet (genesis hash `SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=`): 416002
+* BetaNet (genesis hash `mFgazF+2uRS1tMiL9dsj01hJGySEmPN28B/TjjvpVW0=`): 416003
+
+At the time of writing, these chain IDs do not conflict with any known chain that also uses WalletConnect. In the unfortunate event that this were to happen, the `algorand` query parameter discussed above would be used to differentiate Algorand chains from others.
+
+Future Algorand chains, if introduced, **MUST** be assigned new chain IDs.
+
+Wallets and dApps **MAY** support all of the above chain IDs or only a subset of them. If a chain ID is present to a wallet or dApp that they do not support, they **MUST** terminate the session.
+
+For compatibility with WalletConnect usage prior to this ARC, the following catch-all chain ID is also defined:
+
+* All Algorand Chains (legacy value): 4160
+
+Wallets and dApps **SHOULD** support this chain ID as well for backwards compatibility. Unfortunately this ID alone is not enough to identify which Algorand chain is being used, so extra fields in message requests (i.e. the genesis hash field in a transaction to sign) **SHOULD** be consulted as well to determine this.
+
+### Message Schema
+
+Note: interfaces are defined in TypeScript. These interfaces are designed to be serializable to and from valid JSON objects.
+
+The WalletConnect message schema is a set of [JSON-RPC 2.0](https://www.jsonrpc.org/specification) requests and responses. Decentralized applications will send requests to the wallets and will receive responses as JSON-RPC messages. All requests **MUST** adhere to the following structure:
+
+```typescript
+interface JsonRpcRequest {
+  /**
+   * An identifier established by the Client. Numbers SHOULD NOT contain fractional parts.
+   */
+  id: number;
```

Comment:
Are there any rules to this? I think I've seen its usually epoch in ms?

Reply:
No rules that I'm aware of